### PR TITLE
Queue new scores for processing

### PR DESCRIPTION
**Pull request by nanaya**

### CODE REVIEW

File: app/Models/Solo/Score.php [hunk 1]
```diff
@@ -127,6 +131,32 @@ public function getMode(): string
...
+
+        LaravelRedis::lpush(static::PROCESSING_QUEUE, json_encode([
+            'ProcessHistory' => null,
+            'Score' => [
+                'beatmap_id' => $dataJson['beatmap_id'],
+                'data' => json_encode($dataJson),
+                'id' => $dataJson['id'],
+                'user_id' => $dataJson['user_id'],
+            ],
+            'TotalRetries' => 0,
```
Comment (peppy) [inline app/Models/Solo/Score.php:156]: Not that it really matters, but I think this (and `ProcessHistory`) can be fully omitted.

**Comment by peppy** [comment]
From my end, this can be merged whenever it's ready. Have done very preliminary testing and looks to work as expected.

File: app/Models/Solo/Score.php [hunk 2]
```diff
@@ -127,6 +131,30 @@ public function getMode(): string
         return Beatmap::modeStr($this->ruleset_id);
     }
 
+    /**
+     * Queue the item for score processing
+     *
+     * @param array $dataJson Pre-generated json of the score so it doesn't need to be redundantly generated
+     * @param bool $force By default only newly created scores are queued. Pass true to force queuing the score
+     */
+    public function queueForProcessing(?array $dataJson = null, bool $force = false): void
```
Comment (notbakaneko) [inline app/Models/Solo/Score.php:140]: Is the nullable or flag being used somewhere?

**Review by notbakaneko — CHANGES_REQUESTED** [review]

File: app/Models/Solo/Score.php (hunk 2 above)
Comment (nanaya) [reply app/Models/Solo/Score.php]: this is a weird function and those are originally so its main function (queuing score for processing) can be accessed regardless of state and existence of pre-generated json...

**Review by notbakaneko — APPROVED** [review]

**Pull request merged**